find.go

```diff
@@ -473,18 +473,25 @@ func Nth[T any, N constraints.Integer](collection []T, nth N) (T, error) {
 	return collection[l+n], nil
 }
 
-// Sample returns a random item from collection.
-func Sample[T any](collection []T) T {
+type randomIntGenerator func(int) int
+
+// SampleBy returns a random item from collection, using intFunc as the random index generator.
+func SampleBy[T any](collection []T, randomIntGenerator randomIntGenerator) T {
 	size := len(collection)
 	if size == 0 {
 		return Empty[T]()
 	}
+	return collection[randomIntGenerator(size)]
+}
```

Comment:
I would say you have to fix the comment or the func

```suggestion
// SampleBy returns a random item from collection, using intFunc as the random index generator.
func SampleBy[T any](collection []T, intFunc randomIntGenerator) T {
	size := len(collection)
	if size == 0 {
		return Empty[T]()
	}
	return collection[intFunc(size)]
}
```


```suggestion
// SampleBy returns a random item from collection, using randomIntGenerator as the random index generator.
func SampleBy[T any](collection []T, randomIntGenerator randomIntGenerator) T {
	size := len(collection)
	if size == 0 {
		return Empty[T]()
	}
	return collection[randomIntGenerator(size)]
}
```

Please update SamplesBy accordingly, because right now the function used in it is initFunc

Reply:
Apologies for that! I wanted to push it before having lunch, but didn't properly scan everything.